publish-mqtt/src/homie.rs

```diff
@@ -316,16 +406,22 @@ async fn publish_retained(
 #[cfg(test)]
 mod tests {
     use super::*;
+    use async_channel::Receiver;
 
-    #[tokio::test]
-    #[should_panic(expected = "Tried to add node with duplicate ID")]
-    async fn test_duplicate_node() {
-        let (tx, rx) = async_channel::unbounded();
-        let mut device = HomieDevice::new(
-            tx,
+    fn test_device() -> (HomieDevice, Receiver<Request>) {
```

Comment:
We seem to have gone from the python style of naming (where tests are named `test_*()`) to some other scheme, where fixture factories are called `test_*()`. Is this on purpose?

Maybe the factory could be called `make_test_device()`?

When it comes to naming tests, I quite like the jest style of naming (`describe("the function name", ... it("explodes when you try to add a duplicate node", ...) ...)`). Rust can't be that expressive, but you can get quite close. I quite like the descriptiveness of the suggestions in https://dzone.com/articles/7-popular-unit-test-naming but I disagree with the Super_UglyCased_NamingScheme.

If we go for something like 2, we would end up with tests renamed like this:

(ordering is from the output of `cargo test`. I'm not sure what causes it to run things in that order)
```
add_node => add_node_succeeds_before_and_after_calling_start
ready_before_start => ready_fails_if_called_before_start
read_removed_node => add_node_succeeds_if_node_has_been_removed
duplicate_node => add_node_fails_if_given_duplicate_id
set_firmware => set_firmware_succeeds_before_start
start_empty => start_succeeds_even_with_no_sensors
set_firmware_after_start => set_firmware_fails_after_start
```

Reply:
Naming tests `test_` seemed a bit unnecessary when they are already in a `tests` module. Renamed, mostly as you suggested.